### ISSUE TITLE
3D model is black when rendered...

### ISSUE THREAD

**Issue by keince** [gwaldron/osgearth#2198]
Attached is a 3D model in IVE format that gets rendered black when I load it into my subgraph in our osgEarth application. Lighting is ON and we're using the simply sky Node to provide the lighting.

I've tried running `osgEarth::Registry::shaderGenerator().run(model.get())` on the model node but to no avail. I've also tried using my own shader to get the model to render correctly but that hasn't worked either. Both results yield a black model. If I don't use shaders at all, the model renders white and it is shaded via lighting. There is just no texture and/or color.

Not sure if this is a bug in osgEarth or user error on my part.

The model does render as expected using `osgviewer f35a.ive` but there are no shaders being used.

Any feedback on this would be appreciated. 

[f35a.zip](https://github.com/gwaldron/osgearth/files/10271887/f35a.zip)

**Comment by gwaldron** [comment]
Version? I tried it in an earth file with `osgearth_imgui a.earth --sky`
```
<Map>
    <model>
        <url>D:/data/models/f35a.ive</url>
        <location lat="40.717" long="-74.018" alt="10"/>
    </model>
    <xi:include href="readymap_imagery.xml"/>
    <xi:include href="readymap_elevation.xml"/>
</Map>
```

![image](https://user-images.githubusercontent.com/326618/208768743-dbc7950b-f275-46e5-8fcd-54dbf0e46660.png)

**Comment by keince** [comment]
I'm using osgEarth 3.1. 

I'm not using an earth file because these models are loaded and unloaded dynamically during runtime.

I will try loading the model via an earth file to see if I can see the same results you're seeing.

**Comment by keince** [comment]
Okay. I just added the model via the earth file like what was posted above and the model is still black. The only things that are colored are the lights on the wing tips. The rest of the model is black.

**Comment by leadcoder** [comment]
This maybe related to the same issue i tried to solve last week. My model was also rendered black on intel graphic cards ( nvidia card was ok) when using the imgui-application. However, when using the osgearth_viewer application the intel cards was able to render the model as expected. My conclusion was that when the old gui-controls was enabled  vertexcolor data was passed and this color was reused when my model was rendered. In my case this color was white and the model was rendered as expected. So the problem is that colorarray data is not passed as expeceted, probably related to VertexAttributeAliasing

**Comment by keince** [comment]
@leadcoder That is an interesting analysis. I'm currently running with nvidia and I'm still seeing the problem of a black model. Rather than using our application, I may try loading the earth file in osgearth_viewer to see what the results are.

Thanks for the feedback!

**Comment by keince** [comment]
So...I just loaded our earth file using `osgearth_viewer` built with osgEarth 3.1 and the model looks correct. This suggests that our application is doing something different to render the model black. What that is, escapes me at the moment.

If I find anything, I will post the cause in case anyone else runs into this problem...

**Comment by leadcoder** [comment]
Ok, maybe another issue that you got if you use nvida-hardware, but if you try to load the model in osgearth_viewer be sure that you have some gui-controls visible like the sky-controls or the viewpoints-gui. 

To give some more context regarding my issue: I use osg compiled with osg_OPENGL_PROFILE set to GL2 (windows10) and a custom osgEarth version where compatibility mode was enabled for all osgEarth shaders. When using this setup I'am not forced enable [VertexAttributeAliasing](https://github.com/gwaldron/osgearth/blob/a4a03147f90fb3be43e10a0b76903179a1e9ae8a/src/osgEarth/GLUtils.cpp#L482) 
When I disabled VertexAttributeAliasing the model was rendered correctly (using the osgearth_imgui app) and the vertexcolor data from the model seemed to be transfred into the shader as expected.  

To summarize, my  temporary workaournd is to NOT use VertexAttributeAliasing to get models to render correctly using intel-dirvers on Windows.
